core/store/src/db.rs

```diff
@@ -110,27 +110,60 @@ pub enum DBCol {
+    /// - *Content type*: [near_primitives::sharding::PartialEncodedChunk]
     ColPartialChunks = 14,
     /// Blocks for which chunks need to be applied after the state is downloaded for a particular epoch
-    /// TODO: describe what is exactly inside the rows/cells.
+    /// - *Rows*: BlockHash (CryptoHash)
+    /// - *Content type*: Vec of BlockHash (CryptoHash)
     ColBlocksToCatchup = 15,
     /// Blocks for which the state is being downloaded
+    /// - *Rows*: First block of the epoch (CryptoHash)
+    /// - *Content type*: StateSyncInfo
     ColStateDlInfos = 16,
+    /// Blocks that were ever challenged.
+    /// - *Rows*: BlockHash (CryptoHash)
+    /// - *Content type*: 'true' (bool)
     ColChallengedBlocks = 17,
+    /// Contains all the Shard State Headers.
+    /// - *Rows*: StateHeaderKey (ShardId || BlockHash)
+    /// - *Content type*: ShardStateSyncResponseHeader
     ColStateHeaders = 18,
+    /// Contains all the invalid chunks (that we had trouble decoding or verifying).
+    /// - *Rows*: ShardChunkHeader object
+    /// - *Content type*: EncodedShardChunk
     ColInvalidChunks = 19,
+    /// Contains 'BlockExtra' information that is computed after block was processed.
+    /// Currently it stores only challenges results.
+    /// - *Rows*: BlockHash (CryptoHash)
+    /// - *Content type*: BlockExtra
     ColBlockExtra = 20,
-    /// Store hash of a block per each height, to detect double signs.
+    /// Store hash of all block per each height, to detect double signs.
+    /// - *Rows*: int (height of the block)
+    /// - *Content type*: Map: EpochId -> Set of BlockHash(CryptoHash)
     ColBlockPerHeight = 21,
+    /// Contains State parts that we've received.
+    /// - *Rows*: StatePartKey (BlockHash || ShardId || PartId (u64))
+    /// - *Content type*: state part (bytes)
     ColStateParts = 22,
+    /// Contains mapping from epoch_id to epoch start (first block height of the epoch)
+    /// - *Rows*: EpochId (CryptoHash)  -- TODO: where does the epoch_id come from? it looks like blockHash..
+    /// - *Content type*: BlockHeight (int)
     ColEpochStart = 23,
-    /// Map account_id to announce_account
+    /// Map account_id to announce_account (which peer has announced which account in the current epoch). // TODO: explain account annoucement
+    /// - *Rows*: AccountId (str)
+    /// - *Content type*: AnnounceAccount
     ColAccountAnnouncements = 24,
-    /// Next block hashes in the sequence of the canonical chain blocks
+    /// Next block hashes in the sequence of the canonical chain blocks.
+    /// - *Rows*: BlockHash (CryptoHash)
+    /// - *Content type*: next block: BlockHash (CryptoHash)
     ColNextBlockHashes = 25,
-    /// `LightClientBlock`s corresponding to the last final block of each completed epoch
+    /// `LightClientBlock`s corresponding to the last final block of each completed epoch.
+    /// - *Rows*: EpochId (CryptoHash)
+    /// - *Content type*: LightClientBlockView
```

Comment:
This ... feels fishy. My understanding that `*View` types are for RPC, not for storing directly in DB...

Reply:
I found this code:
https://github.com/near/nearcore/blob/master/chain/chain/src/store.rs#L2603

And this type is BorshSerializable:
https://github.com/near/nearcore/blob/master/core/primitives/src/views.rs#L1433

Reply:
Yeah, it's not a statement about this particular change, it's a statement about pre-existing state of code. 

My understanding is that `*View`-types are intended to be `API` types we use in JSON RPC. So, eg, we use [`Action`](https://github.com/near/nearcore/blob/518685af6a67838e27bf727fd2a677830133f7ad/core/primitives/src/transaction.rs#L48-L66) as internal representation of what we store in the database, and we use [`ActionView`](https://github.com/near/nearcore/blob/518685af6a67838e27bf727fd2a677830133f7ad/core/primitives/src/views.rs#L755-L794) for the RPC (which is reachable from the [`chunk`](https://github.com/near/nearcore/blob/518685af6a67838e27bf727fd2a677830133f7ad/chain/jsonrpc/src/lib.rs#L902-L912) method). 

Some of the `*View` types are not only JSON-encodable, but also borsh-encodable. My understanding that this bottoms out in [`RoutedMessageBody::FinalExecutionOutcomeView`](https://github.com/near/nearcore/blob/518685af6a67838e27bf727fd2a677830133f7ad/chain/network-primitives/src/network_protocol.rs#L181). I am not sure if that should've use `*View` struct or not: it feels weird that [here](https://github.com/near/nearcore/blob/518685af6a67838e27bf727fd2a677830133f7ad/chain/network-primitives/src/types.rs#L325-L330) we have both `*View` and not `*View` struct. So, my current understanding is "some view types are borsh because we use then in p2p messages, which I am not sure why we do". 

What I am noticing here is that there's also at least one `*View` type which we store in the database. And that's surprising, because that means that we store *both* , eg, `ValidatorStakeView` (through `ColEpochLightClientBlocks`) and `ValidatorStake` (through `ColEpochValidatorInfo`) in the database.

Overall, it seems that the codebase as a whole is confused about what `*View` really means, and that some things are miss-classified as `*View`.

Reply:
cc https://github.com/near/nearcore/issues/5516